### PR TITLE
Fix Push service

### PR DESCRIPTION
**Pull request by jaelim-ms**

### CODE REVIEW

File: MobileCenterPush/MobileCenterPushTests/MSPushTests.m [hunk 1]
```diff
@@ -66,18 +66,18 @@ - (void)testInitializationPriorityCorrect {
...
+  [[MSPush sharedInstance] sendPushToken:kMSTestPushToken];
 
-  XCTAssertTrue([MSPush sharedInstance].deviceTokenHasBeenSent);
+  XCTAssertTrue([MSPush sharedInstance].pushTokenHasBeenSent);
 }
 
 - (void)testConvertTokenToString {
   NSString *originalToken = @"563084c4934486547307ea41c780b93e21fe98372dc902426e97390a84011f72";
-  NSData *rawOriginaloken = [MSPushTestUtil convertDeviceTokenToNSData:originalToken];
+  NSData *rawOriginaloken = [MSPushTestUtil convertPushTokenToNSData:originalToken];
```
Comment (bmourat) [inline MobileCenterPush/MobileCenterPushTests/MSPushTests.m]: Small typo here.

Comment (jaelim-ms) [inline MobileCenterPush/MobileCenterPushTests/MSPushTests.m]: Good catch!!

File: MobileCenterPush/MobileCenterPush/MSPush.m [hunk 2]
```diff
@@ -199,17 +199,14 @@ - (void)channel:(id<MSChannel>)channel didFailSendingLog:(id<MSLog>)log withErro
...
-  [MS_USER_DEFAULTS setObject:strDeviceToken forKey:kMSPushServiceStorageKey];
-  [self sendDeviceToken:strDeviceToken];
+  NSString *strPushToken = [self convertTokenToString:pushToken];
+  [MS_USER_DEFAULTS setObject:strPushToken forKey:kMSPushServiceStorageKey];
+  [self sendPushToken:strPushToken];
 }
 
-- (void)application:(UIApplication *)application didFailToRegisterForRemoteNotificationsWithError:(NSError *)error {
-  (void)application;
+- (void)didFailToRegisterForRemoteNotificationsWith:(NSError *)error {
```
Comment (clpolet) [inline MobileCenterPush/MobileCenterPush/MSPush.m]: I am not sure about the naming of the proxied delegate methods, we'll have to discuss about it with PMs probably. But I would still rename this one to `didFailToRegisterForRemoteNotificationsWithError:`.

Comment (jaelim-ms) [inline MobileCenterPush/MobileCenterPush/MSPush.m]: iOS SDK's name is `didFailToRegisterForRemoteNotificationWithError`. Probably we can use the same name for this one.
There is another one which is `didRegisterForRemoteNotificationsWithDeviceToken`.
We don't say `device token` but `push token`. Need to talk to PM for the final names, will discuss that.

**Comment by mh8395648** [comment]
I can not access mobile phone data on this iPhone

File: MobileCenterPush/MobileCenterPush/MSPush.h [hunk 3]
```diff
@@ -9,18 +9,18 @@ NS_ASSUME_NONNULL_BEGIN
 @interface MSPush : MSServiceAbstract
 
 /**
- * Callback for successful registration with device token
+ * Callback for successful registration with push token
```
Comment (TroubleMakerBen) [inline MobileCenterPush/MobileCenterPush/MSPush.h]: We need to finish comments with a "."

File: MobileCenterPush/MobileCenterPush/MSPush.h [hunk 4]
```diff
@@ -9,18 +9,18 @@ NS_ASSUME_NONNULL_BEGIN
...
  *
  * @param deviceToken The device token for remote notifications
```
Comment (TroubleMakerBen) [inline MobileCenterPush/MobileCenterPush/MSPush.h]: We need to finish comments with a "."

File: MobileCenterPush/MobileCenterPush/MSPush.m [hunk 5]
```diff
@@ -28,7 +28,7 @@
 
 @implementation MSPush
 
-@synthesize deviceTokenHasBeenSent;
+@synthesize pushTokenHasBeenSent;
```
Comment (TroubleMakerBen) [inline MobileCenterPush/MobileCenterPush/MSPush.m]: Is the `@synthesize` necessary here?

File: Puppet/NotifExtension/Info.plist [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+	<string>1</string>
+	<key>NSExtension</key>
+	<dict>
+		<key>NSExtensionPointIdentifier</key>
+		<string>com.apple.usernotifications.service</string>
+		<key>NSExtensionPrincipalClass</key>
+		<string>NotificationService</string>
+	</dict>
+</dict>
+</plist>
```
Comment (TroubleMakerBen) [inline Puppet/NotifExtension/Info.plist]: This is probably out of scope of this PR but why not call the `NotifExtension` `NotificationExtension`?

File: MobileCenterPush/MobileCenterPush/MSPush.h (hunk 3 above)
Comment (jaelim-ms) [inline MobileCenterPush/MobileCenterPush/MSPush.h]: Done

File: MobileCenterPush/MobileCenterPush/MSPush.h (hunk 4 above)
Comment (jaelim-ms) [inline MobileCenterPush/MobileCenterPush/MSPush.h]: Done for all Push files.

File: MobileCenterPush/MobileCenterPush/MSPush.m (hunk 5 above)
Comment (jaelim-ms) [inline MobileCenterPush/MobileCenterPush/MSPush.m]: Nope.

File: Puppet/NotifExtension/Info.plist (hunk 6 above)
Comment (jaelim-ms) [inline Puppet/NotifExtension/Info.plist]: It is from Mehrdad. I'll leave it as it is but we can change it in another PR.

**Pull request merged**